services/iam.py

```diff
@@ -2,22 +2,74 @@
 import boto3
 
 
-# client = boto3.client("")
+client = boto3.client("iam")
 
 
 def iam_policy_no_statements_with_admin_access():
+    compliant_resource = []
+    non_compliant_resources = []
+    policies = client.list_policies(Scope="Local")["Policies"]
+
+    for policy in policies:
+        policy_version = client.get_policy_version(PolicyArn=policy["Arn"], VersionId=policy["DefaultVersionId"])[
+            "PolicyVersion"
+        ]
+
+        if "'Effect': 'Allow', 'Action': '*', 'Resource': '*'" not in str(policy_version["Document"]):
+            compliant_resource.append(policy["Arn"])
+        else:
+            non_compliant_resources.append(policy["Arn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resource,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def iam_policy_no_statements_with_full_access():
+    compliant_resource = []
+    non_compliant_resources = []
+    policies = client.list_policies(Scope="Local")["Policies"]
+
+    for policy in policies:
+        policy_version = client.get_policy_version(PolicyArn=policy["Arn"], VersionId=policy["DefaultVersionId"])[
+            "PolicyVersion"
+        ]
+
+        escape = False
+        for statement in policy_version["Document"]["Statement"]:
+            for action in statement["Action"]:
+                if action.endswith(":*"):
+                    non_compliant_resources.append(policy["Arn"])
+                    escape = True
+                    break
+            if escape == True:
+                break
+        else:
+            compliant_resource.append(policy["Arn"])
```

Comment:
각 statement에서 추가로 고려할 사항들이 있습니다. 그에 맞게 수정을 거칠 필요가 있습니다.
- Effect가 Deny인 경우 > 검사 대상이 아님
- Action이 list가 아닌 string일 경우 > character 한글자 씩 반복을 수행

추가로 flag를 제거하기 위해서 list comprehension을 활용할 수도 있습니다.
```suggestion
        for statement in policy_version["Document"]["Statement"]:
            if statement["Effect"] == "Deny":
                continue

            if type(statement["Action"]) == str:
                statement["Action"] = [statement["Action"]]

            full_access_actions = [action for action in statement["Action"] if action.endswith(":*")]
            if full_access_actions:
                non_compliant_resources.append(policy["Arn"])
                break
        else:
            compliant_resource.append(policy["Arn"])
```